Unity-Technologies/ScriptableRenderPipeline
# Only assign the reflection system when an HDRP asset is assigned

### Purpose of this PR
Only assign the reflection system when an HDRP asset is assigned

Fixes 1205109 - custom reflection probe not working when having HDRP code but not using an HDRP asset

---
### Testing status

**Manual Tests**: Manual local testing
Tested to add/remove the HDRP asset in Editor/Play mode to check the reflection probe system is still active in all cases.

**Automated Tests**: -

**Yamato**:-

---
### Comments to reviewers
QA: A more thorough manual testing can be done with different probe types from both legacy and HDRP, and during play mode or editor mode.
In all cases reflection from reflection probes must work
